src/bitcoin-cli.cpp

```diff
@@ -404,7 +404,7 @@ class NetinfoRequestHandler : public BaseRequestHandler
         return UNKNOWN_NETWORK;
     }
     uint8_t m_details_level{0}; //!< Optional user-supplied arg to set dashboard details level
-    bool DetailsRequested() const { return m_details_level > 0 && m_details_level < 5; }
+    bool DetailsRequested() const { return m_details_level; }
```

Comment:
This is better than before, but having a `NETINFO_MAX_LEVEL` and a checking the bounded values in methods and having `DetailsRequested`, but also checking it manually in https://github.com/bitcoin/bitcoin/blob/master/src/bitcoin-cli.cpp#L499-L501 where we're checking the raw `n` value seems quite hacky.

Seems the code is begging for a bounded type - if we're cleaning it up, we might as well try an enum here:

```C++
enum class NetinfoDetailLevel : uint8_t {
    BASIC = 0,
    PEERS,
    PEERS_WITH_ADDR,
    PEERS_WITH_VER,
    FULL
};
```

Reply:
> where we're checking the raw n value

That is to check user input.

I think the abstraction level here is fine, but happy to review if you propose a further abstraction in a follow-up.